src/ucp/wireup/wireup_cm.c

```diff
@@ -836,3 +842,30 @@ ucp_request_t* ucp_ep_cm_close_request_get(ucp_ep_h ep)
 
     return request;
 }
+
+static int ucp_cm_cbs_remove_filter(const ucs_callbackq_elem_t *elem, void *arg)
+{
+    ucp_ep_h ep = arg;
+    ucp_cm_disconnect_progress_arg_t *disconnect_arg;
+    ucp_cm_client_connect_progress_arg_t *client_connect_arg;
+
+    if (elem->cb == ucp_ep_cm_disconnect_progress) {
+        disconnect_arg = elem->arg;
+        return disconnect_arg->ucp_ep == ep;
+    } else if (elem->cb == (ucs_callback_t)ucp_ep_cm_remote_disconnect_progress) {
+        return ep == elem->arg;
+    } else if (elem->cb == ucp_cm_client_connect_progress) {
+        client_connect_arg = elem->arg;
+        return client_connect_arg->ucp_ep == ep;
+    } else if (elem->cb == ucp_cm_server_connect_progress) {
+        return ep == elem->arg;
+    } else {
+        return 0;
+    }
+}
+
+void ucp_ep_cm_slow_cbq_cleanup(ucp_ep_h ep)
```

Comment:
why do we need this wrapper

Reply:
in case when UCP ep is failed during wireup and it's handled (EP is destroyed) from "connect" progress CB and also "disconnect" progress CB might be in the CB queue. I think we can don't check "connect" progress CBs since CB queue is ordered now.

Reply:
so ucp_cm_cbs_remove_filter would be private in file